influence_benchmark/stats/preferences_per_iteration.py

```diff
@@ -8,133 +8,141 @@
 
 def load_trajectories(trajectory_path: Path) -> pd.DataFrame:
     # Read all trajectories from files
-    trajectories = pd.concat([pd.read_json(file, lines=True) for file in trajectory_path.glob("[0-9]*.jsonl")])
+    traj_timesteps = pd.concat([pd.read_json(file, lines=True) for file in trajectory_path.glob("[0-9]*.jsonl")])
 
     # Calculate expected preference
-    trajectories["expected_preference"] = trajectories["preferences"].apply(calculate_expected_preference)
-    return trajectories
+    traj_timesteps["timestep_reward"] = traj_timesteps["preferences"].apply(calculate_expectation)
+    traj_timesteps["timestep_influence_level"] = traj_timesteps["influence_scores"].apply(calculate_expectation)
+    return traj_timesteps
 
 
-def get_top_n_trajectories(trajectory_path: Path, num_chosen_trajectories: int) -> List[Dict]:
-    return get_func_n_trajectories(trajectory_path, num_chosen_trajectories, pd.DataFrame.nlargest)
-
-
-def get_best_worst_n_trajectories(trajectory_path: Path, num_chosen_trajectories: int) -> Tuple[List[Dict], List[Dict]]:
-    top_n = get_func_n_trajectories(
-        trajectory_path, num_chosen_trajectories, pd.DataFrame.nlargest, return_last_turn_only=False
-    )
-    bottom_n = get_func_n_trajectories(
-        trajectory_path, num_chosen_trajectories, pd.DataFrame.nsmallest, return_last_turn_only=False
-    )
+def get_best_worst_n_trajectories(traj_path: Path, num_chosen_trajs: int) -> Tuple[List[Dict], List[Dict]]:
+    top_n = get_func_n_trajectories(traj_path, num_chosen_trajs, pd.DataFrame.nlargest)
+    bottom_n = get_func_n_trajectories(traj_path, num_chosen_trajs, pd.DataFrame.nsmallest)
     return top_n, bottom_n
 
 
 def get_func_n_trajectories(
-    trajectory_path: Path, num_chosen_trajectories: int, func, return_last_turn_only: bool = True
+    trajectory_path: Path, n_chosen_trajs: int, func, return_last_turn_only: bool = False
 ) -> List[Dict]:
     # Load all trajectories from files
-    trajectories = load_trajectories(trajectory_path)
+    trajs = load_trajectories(trajectory_path)
 
     # Add default values for env_name and initial_state_id if not present
-    trajectories["env_name"] = trajectories.get("env_name", "default")
-    trajectories["initial_state_id"] = trajectories.get("initial_state_id", 0)
+    trajs["env_name"] = trajs.get("env_name", "default")
+    trajs["initial_state_id"] = trajs.get("initial_state_id", 0)
```

Comment:
Why do we want to replace `"initial_state_id"` with `0` if it's None? When does this happen? Same with the line above

Reply:
addressed